pharus/component_interface.py

```diff
@@ -314,30 +318,35 @@ def __init__(self, *args, **kwargs):
     def dj_query_route(self):
         fetch_metadata = self.fetch_metadata
         record_header, table_records, total_count = _DJConnector._fetch_records(
-            query=fetch_metadata["query"] & self.restriction[0],
+            query=fetch_metadata["query"] & self.restriction,
             fetch_args=fetch_metadata["fetch_args"],
-            **{
-                k: (
-                    int(v)
-                    if k in ("limit", "page")
-                    else (
-                        v.split(",")
-                        if k == "order"
-                        else json.loads(b64decode(v.encode("utf-8")).decode("utf-8"))
-                    )
-                )
-                for k, v in request.args.items()
-            },
+            limit=int(request.args["limit"]) if "limit" in request.args else 1000,
+            page=int(request.args["page"]) if "page" in request.args else 1,
+            order=request.args["order"].split(",") if "order" in request.args else None,
         )
-        return dict(
-            recordHeader=record_header, records=table_records, totalCount=total_count
+
+        return NumpyEncoder.dumps(
+            dict(
+                recordHeader=record_header,
+                records=table_records,
+                totalCount=total_count,
+            )
         )
 
     def attributes_route(self):
         attributes_meta = _DJConnector._get_attributes(self.fetch_metadata["query"])
-        return dict(
-            attributeHeaders=attributes_meta["attribute_headers"],
-            attributes=attributes_meta["attributes"],
+        query = self.fetch_metadata["query"] & self.restriction[0]
```

Comment:
Shouldn't we have just included this into the above line and return as before? Since now it implements the `dj.U` operation in `_get_attributes`?

```python
attributes_meta = _DJConnector._get_attributes(self.fetch_metadata["query"] & selt.restriction)
```

Reply:
yes that makes sense.